### ISSUE TITLE
Feature Request: Improved Account Settings Multi-Account

### ISSUE THREAD

**Issue by CorneelDragon** [owncloud/android#1693]
### Expected behaviour
Within 'settings' in the iOS-app (multi-account enabled) users see a 'three dot button' next to their name, when they click this a pop-up menu opens with: Edit credentials / Clear Cache / Log Out / Remove Account. See the discussion and mock-ups in: [https://github.com/owncloud/ios/issues/640#issuecomment-220978127](url) 

### Actual behaviour
In the android users still have to press and hold on to their name. In addition, there is no log-out or clear cache option.

**Comment by rperezb** [comment]
@CorneelDragon certainly this requests makes a lot sense so that we keep both clients as similar as possible. For tracking purposes, on the iOS for the moment, we only have 3 options:
- Edit
- Clear cache
- Log out

https://github.com/owncloud/ios/issues/640#issuecomment-221244461

@AndyScherzinger can you help us here? this is mainly a design thing, do you have any idea of how we may include a button to access to this menu instead of using the long press button?
Background, sometimes this menu is not used because users they don't know it's available 😮

**Comment by AndyScherzinger** [comment]
@rperezb sure thing, the drawer branch already implements an owncloud account manager which allows delete and change password where we could also add the clear cache (even though I do not know what that means and if and how this should be triggered, but @davivel might be able to help me out with this.

As for the screenshots on the actual branch for the drawer please see:
* https://github.com/owncloud/android/pull/1559#issuecomment-206977678 drawer with "Manage Accounts"
* https://github.com/owncloud/android/pull/1559#issuecomment-201081043 account manager, at the moment only offering "change password" and remove

I am not sure about the iOS options you mentioned:
* Edit: probably means change password?
* Clear cache: needs clarification
* Log out: does this mean account removed (as done on Android)?

**Comment by rperezb** [comment]
for any reason, #1559 is the one 😄  we do have this pr on the backlog, meaning this it will be included in the short-term

@CorneelDragon will it work this approach for you? it's slighly different to the iOS one:

- When you click on the arrow and extra menu is shown among the options included you may see: "Manage accounts"

![device-2016-06-02-150936](https://cloud.githubusercontent.com/assets/1846927/15745902/7f7a950e-28d4-11e6-87e8-d4021006e8be.png)

If you click on the "Manage accounts" option, you may see:

![device-2016-06-02-150907](https://cloud.githubusercontent.com/assets/1846927/15745877/5d133714-28d4-11e6-9ce3-70d28a9cfd32.png)

There are, for the moment, 2 options:

- the key, for you to update your password
- the x, for you to log out

Is this approach ok for you?

**Comment by rperezb** [comment]
@AndyScherzinger about the Clear cache: this option is a new feature that we have just developen within the iOS app (included in the following version)
Background: limited space on the devices
Goal: when the user clicks on it, all the downloaded files of the selected account are removed locally (just from the device). 
There is one exception, "available offline" data, this data is not removed since the user has selected them  on purpose to have them available on the device.

**Comment by CorneelDragon** [comment]
Actually, I'm a bit confused, when I suggested 'Forget Account' (in the iOS topic), it was disregarded due to the need for consistency with Android and Desktop Clients. Now, I see 'key'-buttons for editing of credentials, a 'cross' for remove accounts, and 'management account' option in Settings, which is also accessible with a 'triangle'-button. 

I have my doubts about the proposed Android UI, just a 'settings'-button (like the 'three dots' in the iOS) with behind that a popup with 'Edit credentials / Clear Cache / Remove / Forget Account' would seem to me more intuitive.

**Comment by AndyScherzinger** [comment]
@CorneelDragon you are right as in inconsistent with the iOS UI but consistent with standard Android UI concepts :)

**Comment by CorneelDragon** [comment]
@AndyScherzinger That could very well be the case (I'm no Android Developer). Nonetheless, from a user point-of-view and if I compare it with the Dropbox-app, I still have my doubts (but of course I respect an expert-decision of your side). 

My point was that in the iOS-thread the decision was made to choose the name 'Remove Account' instead of 'Forget Account' since this was consistent with Android. Now I see that Android will not have any button named 'Remove Account' since this is replaced with a cross. I reckon that an iOS-decision isn't the responsibility of the Android team, but there seems to be some lack of communication (or miscommunication) between the iOS and the Android team (the new iOS version will be sub-optimal in order to be consistent with the old Android version).

**Comment by AndyScherzinger** [comment]
@CorneelDragon no worries :) I still don't understand it :( What do you mean be refering to Dropbox? TO me it seams Dropbox is not multi account capable. So the better choices to take a look at would be the Play store app, GMail, Google Drive, G+, Inbox, ...

> 'management account' option in Settings, which is also accessible with a 'triangle'-button. 
This is a misinterpretation. The triangle just toggles the menus below With this new drawer there won't be any account management in settings anymore but a separated one, just like any Google release app and also like the Android account management itself. The icons might be removed anyways and there will be just a three-dot menu in the account list, since it is likely that there will be more than just two actions. As for consistency it is not necessarily (always) the goal to be consistent UI-wise but platform-consistent.

**Comment by CorneelDragon** [comment]
@AndyScherzinger Thanks for correcting my mis-interpretation :) Your point on Dropbox not supporting multi-acount is fair and, as said, I respect any decision in this regard and I see the importance of being consistent with other Android apps over being consistent with the iOS. 

Regarding my second-point (the decision regarding iOS being consistent with old Android-version) maybe @rperezb can clear this up :)

**Comment by rperezb** [comment]
@CorneelDragon you are right, I missed this pr by the time when we were talking about the change on iOS.
However, we are not only looking for consistency between ownCloud mobile apps but also with the desktop client, ownCloud platform

![screen shot 2016-06-13 at 9 35 28](https://cloud.githubusercontent.com/assets/1846927/16000099/5c654b58-314a-11e6-8f62-dcf2f8792870.png)

Where remove is also used

@dragotin your input?

**Comment by davivel** [comment]
My thoughts:

- Consistency is desirable, not always achievable. Consistency in wording is in general easier to get, but much more difficult in visual design and UI structture, and in this regard we shouldn't enforce it. Platform guidelines are important.

- We need to aim further. Settings should be re-designed so that everything that makes sense to be managed per account is managed per account.

**Comment by CorneelDragon** [comment]
@rperezb Fair enough, (and I don't want to keep bugging about such a small thing). I do hope that if the desktop is updated, 'forget account' will come through. Anyway, thanks for the responses on this and iOS and case be closed :)

**Comment by davivel** [comment]
I wouldn't close, we can take this issue as the starting point for reviewing the layout of settings, having in mind the multiple accounts.

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are
closing it as outdated. If this is still relevant on a current ownCloud release,
please **reopen** (or open a fresh issue) with up-to-date details — thank you\!
